app/assets/javascripts/discourse/app/components/composer-share.js

```diff
@@ -0,0 +1,20 @@
+import Component from "@ember/component";
+import showModal from "discourse/lib/show-modal";
+import { getOwner } from "discourse-common/lib/get-owner";
+
+export default Component.extend({
+  actions: {
+    shareModal() {
+      const composer = getOwner(this).lookup("controller:composer");
```

Comment:
It's better to use:

```js
import { inject as controller } from "@ember/controller";

// then in the component:
composer: controller(),

// and in the action:
this.composer
```

Reply:
Doh, you're right. And you might be onto something with the idea of moving it into the composer controller, since this action extracts a bunch of properties from that controller just to pass it to `showModal`.